### ISSUE TITLE
Using the 'File' option in Gutenberg menu to add a PDF where inline embed is turned off, creates a redundant link

### ISSUE THREAD

**Issue by jhawkmedvesky** [WordPress/gutenberg#64291]
## What problem does this address?
The 'File' option in the Gutenberg menu is used to add a PDF embed to a page, but it can also be used to add a 'Download' button for the PDF, by turning OFF the setting "**Show inline embed**". The accessibility issue here is if you want to use the button option, the result is that it creates 2 links that are exactly the same - one link with the PDF title as the link text, the other link for the button labelled 'Download', and with neither of these links being desirable. In this scenario, I am only interested in displaying a 'Download' button, otherwise I would just create a normal text link, and use the URL of the PDF as the link destination. I know that I have the option to keep the PDF title text and remove the assigned link, allowing the "Download" button to be the only link - however now I am stuck with a button that uses vague link language, and with no PDF title present in the link text, it would be of no value to someone accessing via a screen-reader. I would suggest to use the PDF title as button text to handle that issue, but with many PDF titles being lengthy in nature, using it as the button text is not practical either. 

As a developer, I know how to add the necessary HTML to the button so that I can include non-visible text for the button, but ultimately this will page will be maintained by the client who does know or understand why or how this should be done for PDF Download buttons.

## What is your proposed solution?
I am open to suggestions here, since I don't know if what I consider to be the ideal solution, would be possible. Ultimately there should only be one single link that includes the link's behavior, "Download" followed by the PDF title. To do this you would have to make it so only one single link exists with the button itself - which would still be labelled as "Download", but then would also include non-visible text after it for the PDF Title. This would work for sighted users who can see the title next to the button, as well as screen-readers who would read the button text followed by the (non-visible) PDF title - but the issue there is, at least as far as I know, the only way to have non-visible text is if the theme itself has a custom class built in for that purpose.

Another solution is not ideal, but it might be better than the current outcome with 2 links being created - where just one link is used to wrap around both the PDF title text, and the button with text "Download", however this is not ideal since the link would now contain 2 elements, that when combined and read by a screen-reader might cause confusion, since the word "Download" is now included in the link but it would be announced after the PDF title. This could easily make it sound like the word "Download" is part of the PDF title, and not necessarily the intended behavior of the button.

**Comment by im3dabasia** [comment]
Thank you, @jhawkmedvesky, for raising this issue.

I can confirm that there are two identical links placed next to each other—one displaying the name of the PDF and the other styled as a "Download" button.

The first screenshot shows both links positioned side by side, as described above:

<img width="1467" alt="Image" src="https://github.com/user-attachments/assets/d87f0a96-dadb-42f4-b576-ddb8be77e7e9" />

We can manually modify the name of the download button and remove the redundant link beside it:


<img width="1470" alt="Image" src="https://github.com/user-attachments/assets/27e33d08-e4c4-4a46-a342-d1fe30e10b44" />

The idea here is that while the visual label is correct for sighted users, I see your point about non-sighted users not benefiting from the aria-describedby attribute, which was previously pointing to the PDF name. However, if the button itself includes the name, then a screen reader would announce it properly.



![Image](https://github.com/user-attachments/assets/cb41cf9d-d0b1-4a2e-bea8-13559458e77b)

I believe this approach achieves what you're aiming for. Additionally, adding a toggle to show or hide the link text could give users more control over whether they want to see the text beside the button. However, in this case, the button must have a descriptive name; otherwise, it will be unclear for all users.
